File: contrib/samples/time-tracker-dotnet/TimeTrackerCore/src/TimeTracker/Domain/TimecardImpl.cs
// Name: TimecardImpl.cs
// license-header cs merge-point
//
// This is only generated once! It will never be overwritten.
// You can (and have to!) safely modify it by hand.

using System;

namespace TimeTracker.Domain
{
    /// <summary>
    /// @see TimeTracker.Domain.Timecard
    /// </summary>
    public class TimecardImpl
        : TimeTracker.Domain.Timecard
    {
        /// <summary>
        /// @see TimeTracker.Domain.Timecard#AddTimeAllocation(TimeTracker.Domain.TimeAllocation)
        /// </summary>
        public override void AddTimeAllocation(TimeTracker.Domain.TimeAllocation timeAllocation)
        {
            Allocations.Add(timeAllocation);
            timeAllocation.Timecard = this;
        }
    }
}

File: documentation/samples/time-tracker-java/core/src/java/org/andromda/timetracker/domain/TimecardImpl.java
// license-header java merge-point
/**
 * This is only generated once! It will never be overwritten.
 * You can (and have to!) safely modify it by hand.
 */
package org.andromda.timetracker.domain;

/**
 * @see org.andromda.timetracker.domain.Timecard
 */
public class TimecardImpl
    extends org.andromda.timetracker.domain.Timecard
{
    /**
     * The serial version UID of this class. Needed for serialization.
     */
    private static final long serialVersionUID = 4974757278671085699L;

    /**
     * @see org.andromda.timetracker.domain.Timecard#addTimeAllocation(org.andromda.timetracker.domain.TimeAllocation)
     */
    public void addTimeAllocation(org.andromda.timetracker.domain.TimeAllocation timeAllocation)
    {
        getAllocations().add(timeAllocation);
        timeAllocation.setTimecard(this);
    }
}